Description: Update the Autoconf configure.in template.
  Use the two-argument form of AC_INIT and AC_CONFIG_SRCDIR and
  update/replace the references to PACKAGE and VERSION.
  .
  Replace AC_CONFIG_HEADER by AC_CONFIG_HEADERS.
  .
  Use the three-argument form of AC_DEFINE.
  .
  In mbuffer.c, move the block that undefines HAVE_SENDFILE until
  after the last possible place where config.h may be included
  again to avoid it accidentally redefining it.
Author: Peter Pentchev <roam@ringlet.net>
Forwarded: yes
Last-Update: 2014-10-22

--- a/configure.in
+++ b/configure.in
@@ -1,12 +1,10 @@
-AC_INIT(mbuffer.c)
-AC_CONFIG_HEADER(config.h)
+AC_INIT([mbuffer], [20140310])
+AC_CONFIG_HEADERS([config.h])
+AC_CONFIG_SRCDIR([mbuffer.c])
 
 AC_CANONICAL_SYSTEM
 AC_EXEEXT
 
-PACKAGE=mbuffer
-VERSION=20140310
-
 if test "${LD_LIBRARY_PATH}" != "" ; then
 	echo ""
 	echo "    WARNING: LD_LIBRARY_PATH is set. This can cause severe problems!!!"
@@ -87,14 +85,14 @@
 AC_SEARCH_LIBS(clock_gettime,rt,,AC_MSG_ERROR(could not find the library containing clock_gettime!))
 AC_SEARCH_LIBS(sem_init,rt posix,,AC_MSG_ERROR(could not find semaphore functions!))
 AC_SEARCH_LIBS(rint,m,,AC_MSG_ERROR(unable to find rint))
-AC_SEARCH_LIBS(seteuid,c,AC_DEFINE(HAVE_SETEUID),)
-AC_SEARCH_LIBS(atoll,c,,AC_DEFINE_UNQUOTED(atoll,atol))
-AC_SEARCH_LIBS(hstrerror,resolv socket nsl,AC_DEFINE(HAVE_HSTRERROR),)
-AC_SEARCH_LIBS(getaddrinfo,resolv socket nsl,AC_DEFINE(HAVE_GETADDRINFO),)
+AC_SEARCH_LIBS(seteuid,c,AC_DEFINE([HAVE_SETEUID], [1], [Define to 1 if you have the seteuid() function]),)
+AC_SEARCH_LIBS(atoll,c,,AC_DEFINE([atoll],[atol],[Define to 1 if you do not have the atoll() function to use atol() instead]))
+AC_SEARCH_LIBS(hstrerror,resolv socket nsl,AC_DEFINE([HAVE_HSTRERROR], [1], [Define to 1 if you have the hstrerror() function]),)
+AC_SEARCH_LIBS(getaddrinfo,resolv socket nsl,AC_DEFINE([HAVE_GETADDRINFO], [1], [Define to 1 if you have the getaddrinfo() function]),)
 AC_SEARCH_LIBS(socket,socket,,AC_MSG_ERROR(could not find the library containing network functions!))
 AC_SEARCH_LIBS(gethostbyname,nsl,,AC_MSG_ERROR(could not find the library containing name convertion functions!))
-AC_SEARCH_LIBS(sendfile,sendfile,AC_DEFINE(HAVE_SENDFILE,1,libsendfile is available),)
-AC_CHECK_HEADER(sys/sendfile.h,AC_DEFINE(HAVE_SENDFILE_H,1,sendfile in sendfile.h),)
+AC_SEARCH_LIBS(sendfile,sendfile,AC_DEFINE([HAVE_SENDFILE],[1],[libsendfile is available]),)
+AC_CHECK_HEADER(sys/sendfile.h,AC_DEFINE([HAVE_SENDFILE_H],[1],[sendfile in sendfile.h]),)
 AC_FUNC_ALLOCA
 
 
@@ -103,7 +101,7 @@
         ,enable_debug=yes
         )
 if test x$enable_debug = xyes ; then
-	AC_DEFINE_UNQUOTED(DEBUG)
+	AC_DEFINE([DEBUG], [1], [Define to 1 to build in debugging mode])
 fi
 
 AC_ARG_ENABLE(md5,
@@ -111,9 +109,9 @@
 	,enable_md5=auto
 	)
 if test x$enable_md5 != xno ; then
-	AC_SEARCH_LIBS(mhash_init,mhash,AC_DEFINE_UNQUOTED(HAVE_LIBMHASH),
-	AC_SEARCH_LIBS(MD5Init,md5,AC_DEFINE_UNQUOTED(HAVE_LIBMD5),
-	AC_SEARCH_LIBS(MD5_Init,crypto,AC_DEFINE_UNQUOTED(HAVE_LIBCRYPTO),
+	AC_SEARCH_LIBS(mhash_init,mhash,AC_DEFINE([HAVE_LIBMHASH], [1], [Define to 1 if you have the mhash library]),
+	AC_SEARCH_LIBS(MD5Init,md5,AC_DEFINE([HAVE_LIBMD5], [1], [Define to 1 if you have the md5 library]),
+	AC_SEARCH_LIBS(MD5_Init,crypto,AC_DEFINE([HAVE_LIBCRYPTO], [1], [Define to 1 if you have the OpenSSL crypto library]),
 		if test x$enable_md5 = xauto ; then
 			AC_MSG_WARN(could not find a library containing the md5 hash functions - md5 support disabled)
 		else
@@ -123,10 +121,6 @@
 fi
 
 AC_SUBST(DEBUG)
-AC_DEFINE_UNQUOTED(PACKAGE,"${PACKAGE}")
-AC_DEFINE_UNQUOTED(VERSION,"${VERSION}")
-AC_SUBST(PACKAGE)
-AC_SUBST(VERSION)
 AC_SUBST(prefix)
 AC_SUBST(AUTOCONF)
 AC_DEFINE(_REENTRANT,1,Needed for thread safe compilation)
--- a/mbuffer.1.in
+++ b/mbuffer.1.in
@@ -1,4 +1,4 @@
-.TH "mbuffer" "1" "@VERSION@" "Thomas Maier-Komor" "console utility"
+.TH "mbuffer" "1" "@PACKAGE_VERSION@" "Thomas Maier-Komor" "console utility"
 .SH "NAME"
 mbuffer \- measuring buffer
 .SH "SYNTAX"
--- a/mbuffer.c
+++ b/mbuffer.c
@@ -58,14 +58,6 @@
 #ifdef HAVE_SENDFILE_H
 #include <sys/sendfile.h>
 #endif
-/* if this sendfile implementation does not support sending from buffers,
-   disable sendfile support */
-	#ifndef SFV_FD_SELF
-	#ifdef __GNUC__
-	#warning sendfile is unable to send from buffers
-	#endif
-	#undef HAVE_SENDFILE
-	#endif
 #endif
 
 #ifndef EBADRQC
@@ -122,6 +114,15 @@
 #include "network.h"
 #include "log.h"
 
+/* if this sendfile implementation does not support sending from buffers,
+   disable sendfile support */
+#ifndef SFV_FD_SELF
+#ifdef __GNUC__
+#warning sendfile is unable to send from buffers
+#endif
+#undef HAVE_SENDFILE
+#endif
+
 char
 	*Prefix;
 int 
@@ -1349,7 +1350,7 @@
 static void version(void)
 {
 	(void) fprintf(stderr,
-		"mbuffer version "VERSION"\n"\
+		"mbuffer version "PACKAGE_VERSION"\n"\
 		"Copyright 2001-2014 - T. Maier-Komor\n"\
 		"License: GPLv3 - see the LICENSE file\n"\
 		"This program comes with ABSOLUTELY NO WARRANTY!!!\n"
